devito/dle/backends/basic.py

```diff
@@ -3,15 +3,15 @@
 import cgen as c
 import numpy as np
 
-from devito.dimension import IncrDimension
 from devito.dle.backends import AbstractRewriter, dle_pass, complang_ALL
 from devito.ir.iet import (Denormals, Call, Callable, List, ArrayCast,
                            Transformer, FindSymbols, retrieve_iteration_tree,
                            filter_iterations, derive_parameters)
 from devito.parameters import configuration
 from devito.symbolics import as_symbol
 from devito.tools import flatten
-from devito.types import Scalar
+from devito.types.basic import Scalar
```

Comment:
from devito.types import Scalar, IncrDimension

Reply:
Ok that's like the 3rd time I've been asked to change import back and forth between absolute and not absolute path. `Scalar` not exported in types so absolute, I'm not respending  another 3h redoing all imports again.

Reply:
clearly I'm not that crazy

the absolute ones are from when you are **within** types . All other ones are partial imports. I wrote it initially at some point. I can fix it though, just tell me when I can touch the branch

Reply:
That's what I did before and you said now it export too many internal types in `_all__`, and that's the only way without exporting it. If you have a way go for it.